src/pushjack/apns.py

```diff
@@ -764,6 +776,346 @@ class APNSExpiredToken(namedtuple('APNSExpiredToken', ['token', 'timestamp'])):
+            action_loc_key (str, optional): Display an alert that includes the
+                ``Close`` and ``View`` buttons. The string is used as a key to
+                get a localized string in the current localization to use for
+                the right button’s title instead of ``"View"``.
+            loc_key (str, optional): A key to an alert-message string in a
+                ``Localizable.strings`` file for the current localization.
+            loc_args (list, optional): List of string values to appear in place
+                of the format specifiers in ``loc_key``.
+            launch_image (str, optional): The filename of an image file in the
+                app bundle; it may include the extension or omit it.
+            mutable_content (bool, optional): if ``True``, triggers Apple
+                Notification Service Extension. Defaults to ``None``.
+            thread_id (str, optional): Identifier for grouping notifications.
+                iOS groups notifications with the same thread identifier
+                together in Notification Center. Defaults to ``None``.
+            extra (dict, optional): Extra data to include with the alert.
+        """
+        assert device_id, 'Must provide device IDs.'
+
+        if max_payload_length is None:
+            max_payload_length = self.default_max_payload_length
+
+        message = APNSMessage(message,
+                              max_payload_length=max_payload_length,
+                              **options)
+
+        validate_tokens([device_id])
+        validate_message(message)
+
+        return self._send_message(
+            device_id=device_id,
+            message=message,
+            low_priority=low_priority,
+            expiration=expiration)
+
+    def send_messages(self,
+                      device_ids,
+                      message=None,
+                      expiration=None,
+                      low_priority=None,
+                      max_payload_length=None,
+                      **options):
+        unsendable_devices = []
+        if max_payload_length is None:
+            max_payload_length = self.default_max_payload_length
+
+        message = APNSMessage(message,
+                              max_payload_length=max_payload_length,
+                              **options)
+
+        validate_tokens(device_ids)
+        validate_message(message)
+
+        for device_id in device_ids:
+
+            try:
+                self._send_message(
+                    device_id=device_id, message=message,
+                    low_priority=low_priority, expiration=expiration)
+            except Exception:
+                unsendable_devices.append(device_id)
+
+        if not unsendable_devices:
+            return
+
+        if len(unsendable_devices) != len(device_ids):
+            raise APNSMessagePartiallySentError
+
+        raise APNSUnsendableError
+
+    def _send_message(
```

Comment:
Would also want to keep this inline with renaming `send_message` to `send` and perhaps call it `_send`.

Reply:
OK, sure.